src/common/Gateway.sol

```diff
@@ -353,22 +357,19 @@ contract Gateway is Auth, IGateway, Recoverable {
     {
         bytes memory proof = processor.createMessageProof(payload);
 
-        uint256 proofGasLimit = gasService.gasLimit(centrifugeId, proof);
-
-        uint256 messageGasLimit = 0;
+        uint256 gasLimit = 0;
         for (uint256 pos; pos < payload.length;) {
             bytes calldata inner = payload[pos:payload.length];
-            messageGasLimit += gasService.gasLimit(centrifugeId, inner);
+            gasLimit += messageGasLimit;
             pos += processor.messageLength(inner);
         }
 
         perAdapter = new uint256[](adapters[centrifugeId].length);
```

Comment:
@hieronx I think returning this `perAdapter` in `estimate()` now does not add much info. Just a slight variation due to the Adapter in comparison with the big value `messageGasLimit`. I'm in favor of removing this and simplifying tests regarding this WDYT?

Reply:
Yeah think that is fine now!